src/Microsoft.Extensions.Logging/Logger.cs

```diff
@@ -96,10 +108,16 @@ public bool IsEnabled(LogLevel logLevel)
 
         public IDisposable BeginScopeImpl(object state)
         {
+            if (_loggers == null)
+            {
+                return null;
```

Comment:
When `_loggers == null` return a singleton IDisposable object which does nothing in the Dispose method. No allocations, but BeginScopeImpl can always returned a non-null value.

Also when `_loggers.Count == 1` you can `return _loggers[0].BeginScopeImpl(state)` - there's no reason to create an aggregate disposable around a single instance